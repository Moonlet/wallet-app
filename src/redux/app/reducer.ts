import { IAppState } from './state';
import { IAction } from '../types';
import {
    SET_ACCEPTED_TC_VERSION,
    SHOW_HINT,
    RESET_FAILED_LOGINS,
    INCREMENT_FAILED_LOGINS,
    SET_APP_BLOCK_UNTIL,
    RESET_ALL_DATA,
    REDUX_UPDATE_STATE
} from './actions';

const intialState: IAppState = {
    version: 1,
    tcAcceptedVersion: undefined,
    hints: {
        SEND_SCREEN: {
            ADDRESS_BOOK: 0
        },
        WALLETS_SCREEN: {
            WALLETS_LIST: 0
        },
        MANAGE_ACCOUNT: {
            TOKENS_LIST: 0
        }
    },
    failedLogins: 0,
    blockUntil: undefined
};

export default (state: IAppState = intialState, action: IAction): IAppState => {
    switch (action.type) {
        case SET_ACCEPTED_TC_VERSION:
            return {
                ...state,
                tcAcceptedVersion: action.data
            };

        case SHOW_HINT:
            return {
                ...state,
                hints: {
                    ...state.hints,
                    [action.data.screen]: {
                        [action.data.component]:
                            state.hints[action.data.screen][action.data.component] + 1
                    }
                }
            };

        case RESET_FAILED_LOGINS:
            return {
                ...state,
                failedLogins: 0
            };

        case INCREMENT_FAILED_LOGINS:
            return {
                ...state,
                failedLogins: state.failedLogins + 1
            };

        case SET_APP_BLOCK_UNTIL:
            return {
                ...state,
                blockUntil: action.data.date
            };

        case RESET_ALL_DATA:
            return intialState;

        case REDUX_UPDATE_STATE: {
            return {
                ...state,
                version: action.data.state.app.version
            };
        }

        default:
            break;
    }
    return state;
};
