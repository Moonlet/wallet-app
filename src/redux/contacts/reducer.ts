import { IAction } from '../types';
import { IContactsState } from './state';
import { CONTACT_ADD, CONTACT_DELETE, CONTACT_UPDATE_NAME } from './actions';

const intialState: IContactsState = {};

export default (state: IContactsState = intialState, action: IAction) => {
    switch (action.type) {
        case CONTACT_ADD:
            return {
                ...state,
                [action.data.address]: action.data
            };

        case CONTACT_DELETE:
            return Object.keys(state)
                .filter(key => key !== action.data.address)
                .reduce((result, current) => {
                    result[current] = state[current];
                    return result;
                }, {});

        case CONTACT_UPDATE_NAME:
            const newState = state;
            newState[action.data.address] = action.data;
            return newState;

        default:
            break;
    }
    return state;
};
