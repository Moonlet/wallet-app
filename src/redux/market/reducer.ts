import { IAction } from '../types';

const initialState = {
    usdPrices: {
        USD: 1,
        BTC: 8349.75,
        ETH: 182.25,
        ZIL: 0.005821,
        ATOM: 3.03,
        XLM: 0.06122
    },
    change: {
        daily: {
            USD: {
                USD: 0,
                BTC: 0.0045,
                ETH: -0.0016
            },
            BTC: {
                USD: 0.0043,
                BTC: 0,
                ETH: -0.0016
            },
            ETH: {
                USD: 0.0143,
                BTC: 0.0045,
                ETH: 0
            },
            ZIL: {
                USD: 0.0002,
                BTC: 0.0045,
                ETH: 0.0216
            },
            ATOM: {
                USD: 0.0045,
                BTC: -0.0045,
                ETH: -0.0016
            },
            XLM: {
                USD: 0.0045,
                BTC: 0.0033,
                ETH: -0.0016
            }
        }
    }
};

const intialState: any = initialState;

export default (state: any = intialState, action: IAction): any => {
    return state;
};
