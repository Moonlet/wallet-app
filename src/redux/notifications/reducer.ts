import { INotificationsState } from './state';
import { IAction } from '../types';
import { SET_UNSEEN_NOTIFICATIONS, SET_NOTIFICATIONS, MARK_SEEN } from './actions';

const initialState: INotificationsState = {
    unseenNotifications: 0,
    notifications: {}
};

export default (
    state: INotificationsState = initialState,
    action: IAction
): INotificationsState => {
    switch (action.type) {
        case SET_UNSEEN_NOTIFICATIONS:
            return {
                ...state,
                unseenNotifications: action.data.unseenNotifications
            };

        case SET_NOTIFICATIONS: {
            let notifications = {};

            action.data.notifications.map((notif: any) => {
                const notifData = {
                    walletId: notif.walletId,
                    title: notif.title,
                    body: notif.body,
                    seen: notif.seen,
                    data: notif.data
                };

                notifications = {
                    ...notifications,
                    [notif.data.blockchain]: {
                        ...notifications[notif.data.blockchain],
                        [notif._id]: notifData
                    }
                };
            });

            return {
                ...state,
                notifications
            };
        }

        case MARK_SEEN:
            return {
                ...state,
                notifications: {
                    ...state.notifications,
                    [action.data.blockchain]: {
                        ...state.notifications[action.data.blockchain],
                        [action.data.notifId]: {
                            ...state.notifications[action.data.blockchain][action.data.notifId],
                            seen: true
                        }
                    }
                }
            };

        default:
            break;
    }
    return state;
};
