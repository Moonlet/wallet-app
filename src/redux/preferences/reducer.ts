import { IPrefState } from './state';
import { IAction } from '../types';
import {
    PREF_SET_CURRENCY,
    TOGGLE_TOUCH_ID,
    SET_TEST_NET,
    PREF_SET_NETWORK_TEST_NET_CHAIN_ID,
    PREF_SET_BLOCKCHAIN_ACTIVE_STATE,
    PREF_SET_BLOCKCHAIN_ORDER,
    PREF_SET_DEVICE_ID
} from './actions';

const initialState: IPrefState = {
    currency: 'USD',
    testNet: false,
    touchID: false,
    networks: {},
    blockchains: {},
    deviceId: ''
};

export default (state: IPrefState = initialState, action: IAction): IPrefState => {
    switch (action.type) {
        case PREF_SET_NETWORK_TEST_NET_CHAIN_ID:
            return {
                ...state,
                networks: {
                    ...state.networks,
                    [action.data.blockchain]: {
                        ...state.networks[action.data.blockchain],
                        testNet: action.data.chainId
                    }
                }
            };
        case PREF_SET_BLOCKCHAIN_ACTIVE_STATE:
            return {
                ...state,
                blockchains: {
                    ...state.blockchains,
                    [action.data.blockchain]: {
                        ...state.blockchains[action.data.blockchain],
                        active: action.data.active
                    }
                }
            };
        case PREF_SET_BLOCKCHAIN_ORDER:
            return {
                ...state,
                blockchains: action.data.blockchains
            };
        case SET_TEST_NET:
            return {
                ...state,
                testNet: !state.testNet
            };
        case TOGGLE_TOUCH_ID:
            state = { ...state };
            state.touchID = !state.touchID;
            break;
        case PREF_SET_CURRENCY:
            return {
                ...state,
                currency: action.data.currency
            };
            break;
        case PREF_SET_DEVICE_ID:
            return {
                ...state,
                deviceId: action.data.deviceId
            };
        default:
            break;
    }
    return state;
};
