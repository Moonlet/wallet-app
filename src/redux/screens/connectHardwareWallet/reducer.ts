import { IConnectHardwareWalletState } from './state';
import { IAction } from '../../types';
import {
    VERIFY_ADDRESS_ON_DEVICE,
    FEATURE_NOT_SUPPORTED,
    TO_INITIAL_STATE,
    CONNECT_IN_PROGRESS
} from './actions';

const intialState: IConnectHardwareWalletState = {
    verifyAddress: false,
    featureNotSupported: false,
    connectInProgress: false
};

export default (
    state: IConnectHardwareWalletState = intialState,
    action: IAction
): IConnectHardwareWalletState => {
    switch (action.type) {
        case VERIFY_ADDRESS_ON_DEVICE:
            return { ...state, verifyAddress: action.data };
        case FEATURE_NOT_SUPPORTED:
            return { ...state, featureNotSupported: true };
        case CONNECT_IN_PROGRESS:
            return { ...state, connectInProgress: true };
        case TO_INITIAL_STATE:
            return { ...state, featureNotSupported: false, verifyAddress: false };
        default:
            break;
    }
    return state;
};
