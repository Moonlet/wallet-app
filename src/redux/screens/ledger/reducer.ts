import { ILedgerState } from './state';
import { IAction } from '../../types';
import { VERIFY_ADDRESS_ON_DEVICE } from './actions';

const intialState: ILedgerState = {
    verifyAddress: false
};

export default (state: ILedgerState = intialState, action: IAction): ILedgerState => {
    switch (action.type) {
        case VERIFY_ADDRESS_ON_DEVICE:
            return { ...state, verifyAddress: action.data };

        default:
            break;
    }
    return state;
};
