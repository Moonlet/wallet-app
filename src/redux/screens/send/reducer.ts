import { ISendScreenState } from './state';
import { IAction } from '../../types';
import { REVIEW_TRANSACTION } from './actions';

const intialState: ISendScreenState = {
    reviewTransaction: false
};

export default (state: ISendScreenState = intialState, action: IAction): ISendScreenState => {
    switch (action.type) {
        case REVIEW_TRANSACTION:
            return { ...state, reviewTransaction: action.data };

        default:
            break;
    }
    return state;
};
