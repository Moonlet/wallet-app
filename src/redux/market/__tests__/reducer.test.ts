import reducer from '../reducer';

describe('market reducer', () => {
    test('should set initial state', () => {
        expect(
            reducer(undefined as any, {
                type: '',
                data: ''
            })
        ).toMatchSnapshot();
    });
});
