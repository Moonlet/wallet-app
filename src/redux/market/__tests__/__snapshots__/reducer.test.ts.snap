// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`market reducer should set initial state 1`] = `
Object {
  "change": Object {
    "daily": Object {
      "ATOM": Object {
        "BTC": -0.0045,
        "ETH": -0.0016,
        "USD": 0.0045,
      },
      "BTC": Object {
        "BTC": 0,
        "ETH": -0.0016,
        "USD": 0.0043,
      },
      "ETH": Object {
        "BTC": 0.0045,
        "ETH": 0,
        "USD": 0.0143,
      },
      "USD": Object {
        "BTC": 0.0045,
        "ETH": -0.0016,
        "USD": 0,
      },
      "XLM": Object {
        "BTC": 0.0033,
        "ETH": -0.0016,
        "USD": 0.0045,
      },
      "ZIL": Object {
        "BTC": 0.0045,
        "ETH": 0.0216,
        "USD": 0.0002,
      },
    },
  },
  "exchangeRates": Object {
    "BAT": Object {
      "ETH": 0.001281,
    },
    "CENNZ": Object {
      "ETH": 0.0005804,
    },
    "CRO": Object {
      "ETH": 0.0002848,
    },
    "ETH": Object {
      "BTC": 0.017713,
      "CNY": 1032.85,
      "DAI": 151.15,
      "EUR": 135.54,
      "JPY": 16481.83,
      "SGD": 202.43,
      "USD": 150.76,
      "USDT": 150.85,
    },
    "HEDG": Object {
      "ETH": 0.01471,
    },
    "HT": Object {
      "ETH": 0.02053,
    },
    "KCS": Object {
      "ETH": 0.00701,
    },
    "LEO": Object {
      "ETH": 0.005713,
    },
    "LINK": Object {
      "ETH": 0.01501,
    },
    "MKR": Object {
      "ETH": 3.246,
    },
    "OKB": Object {
      "ETH": 0.01938,
    },
    "PAX": Object {
      "ETH": 0.006644,
    },
    "SNX": Object {
      "ETH": 0.006693,
    },
    "SXP": Object {
      "ETH": 0.009383,
    },
    "USDC": Object {
      "ETH": 0.006639,
    },
    "ZIL": Object {
      "CNY": 0.03273,
      "DAI": 0.004802,
      "EUR": 0.004293,
      "JPY": 0.5241,
      "SGD": 0.006416,
      "USD": 0.004778,
      "USDT": 0.004723,
    },
    "ZRX": Object {
      "ETH": 0.001475,
    },
  },
}
`;
